tests/framework/extensions/kubeapi/rbac/list.go

```diff
@@ -62,3 +63,58 @@ func ListClusterRoleBindings(client *rancher.Client, clusterName string, listOpt
 
 	return crbList, nil
 }
+
+// ListGlobalRoleBindings is a helper function that uses the dynamic client to list globalrolebindings from local cluster.
+// ListGlobalRoleBindings accepts ListOptions for specifying desired parameters for listed objects.
+func ListGlobalRoleBindings(client *rancher.Client, clusterName string, listOpt metav1.ListOptions) (*v3.GlobalRoleBindingList, error) {
```

Comment:
I'm not super familiar with your framework, but I would recommend trying to use a typed lasso controller rather than needing to go from unstructured -> structured. It would be a bit easier to use and not require functions like this. I've done something similar in the past, see the following:
- [Building the controller for a rancher CRD](https://github.com/rancher/csp-adapter/blob/065a9f42407aed8decf10c82d96325e02445ee60/pkg/clients/k8s/client.go#L68-L95)
- [Using the client from the controller](https://github.com/rancher/csp-adapter/blob/065a9f42407aed8decf10c82d96325e02445ee60/pkg/clients/k8s/client.go#L229)

Wrangler may also offer easier ways to do this as well, though I'm a bit less familiar with it.

Reply:
Hi Michael the reason for this is the built in session clean up we have with the dynamic client. In this case it won't really be an issue, but we do indeed have in our immediate backlog do incorporate the wrangler controllers/clients to make this a much easier process. No strong opinion either way, but yes it is on our radar especially with all the new public api test automation coming in.